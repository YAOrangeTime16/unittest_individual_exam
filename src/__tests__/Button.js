import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Button from '../components/Button';

test('the button should show a proper class name', ()=>{
    const wrapper = shallow((
        <Button onClick={()=>{}} danger={true}>
         <div>testing</div>
        </Button>
    ));
    expect(wrapper.find('.bg-red-dark').exists()).toBeTruthy();
});