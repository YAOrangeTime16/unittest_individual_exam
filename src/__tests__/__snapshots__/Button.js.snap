// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`danger = false 1`] = `
<button
  className="bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full "
  data-test="button"
  onClick={[Function]}
>
  button title
</button>
`;

exports[`danger = true 1`] = `
<button
  className="bg-red-dark hover:bg-red-darker text-white font-bold py-2 px-4 rounded-full "
  data-test="button"
  onClick={[Function]}
>
  button title
</button>
`;
